src/compile/repeat.ts

```diff
@@ -0,0 +1,204 @@
+    super(spec, parent, parentGivenName, config);
+
+    this.repeat = spec.repeat;
+    this.children = this._initChildren(spec, this.repeat, repeatValues, config);
+  }
+
+  private _initChildren(spec: RepeatSpec, repeat: Repeat, repeatValues: RepeatValues, config: Config): Model[] {
+    const children: Model[] = [];
+    const row = repeat.row || [repeatValues ? repeatValues.row : null];
+    const column = repeat.column || [repeatValues ? repeatValues.column : null];
+
+    // cross product
+    for (const r in row) {
+      if (row.hasOwnProperty(r)) {
+        const rowField = row[r];
+        for (const c in column) {
+          if (column.hasOwnProperty(c)) {
+            const columnField = column[c];
+
+            const name = (rowField ? '_' + rowField : '') + (columnField ? '_' + columnField : '');
+
+            const childRepRepeatValues = {
+              row: rowField,
+              column: columnField
+            };
+
+            children.push(buildModel(duplicate(spec.spec), this, this.getName('child' + name), childRepRepeatValues, config));
+          }
+        }
+      }
+    }
+
+    return children;
+  }
+
+  public channelHasField(channel: Channel): boolean {
+    // repeat does not have any channels
+    return false;
+  }
+
+  public parseData() {
+    this.children.forEach((child) => {
+      child.parseData();
+    });
+  }
+
+  public parseSelection() {
+    // TODO: @arvind can write this
+    // We might need to split this into compileSelectionData and compileSelectionSignals?
+  }
+
+  public parseLayoutData() {
+    this.children.forEach(child => {
+      child.parseLayoutData();
+    });
+    this.component.layout = {
+      width: null,
+      height: null
+    };
+  }
+
+  public parseScale(this: RepeatModel) {
+    const model = this;
+
+    this.component.scales = {};
+
+    this.children.forEach(function(child) {
+      child.parseScale();
+    });
+  }
+
+  public parseMark() {
+    for (const child of this.children) {
+      child.parseMark();
+    }
+  }
+
+  public parseAxis() {
+    for (const child of this.children) {
+      child.parseAxis();
+    }
+  }
+
+  public parseAxisGroup(): void {
+    return null;
+  }
+
+  public parseLegend() {
+    for (const child of this.children) {
+      child.parseLegend();
+    }
+  }
+
+  public assembleData(): VgData[] {
+     if (!this.parent) {
+      // only assemble data in the root
+      return assembleData(vals(this.component.data.sources));
+    }
+
+    return [];
+  }
+
+  public assembleParentGroupProperties(cellConfig: CellConfig): any {
+    return null;
+  }
+
+  public assembleSignals(signals: any[]): any[] {
+    return [];
+  }
+
+  public assembleSelectionData(data: VgData[]): VgData[] {
+    return [];
+  }
+
+  public assembleScales(): VgScale[] {
+    // combine with scales from children
+    return this.children.reduce((scales, c) => {
```

Comment:
Is this where you ask whether we need scale merging? 

Now I wonder what happens if we add color to each of them.